### PR TITLE
Fix o1-preview model support in Azure OpenAI

### PR DESCRIPTION
**Pull request by henryperkins**
Related to #903

Update code to support Azure OpenAI `o1-preview` model with `maxCompletionTokens` property.

* Add `maxCompletionTokens` property to `ModelConfig` interface in `src/aiParams.ts`.
* Update `handleOpenAIExtraArgs` method in `src/LLMProviders/chatModelManager.ts` to set `maxCompletionTokens` property for `o1-preview` model.
* Pass `maxCompletionTokens` property to `ChatOpenAI` constructor in `getModelConfig` method in `src/LLMProviders/chatModelManager.ts`.
* Update `createChainWithNewModel` and `setChain` methods in `src/LLMProviders/chainManager.ts` to handle `o1-preview` model.
* Add specific mention of `o1-preview` model support in Azure OpenAI section in `README.md`.
* Add instructions for setting up `o1-preview` model in Azure OpenAI section in `src/settings/components/ApiSettings.tsx`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/logancyang/obsidian-copilot/pull/904?shareId=1490d18b-a3f2-41e7-9604-5dd009eb9941).

### CODE REVIEW

**Comment by logancyang** [comment]
Looks like code duplication with #850

File: src/LLMProviders/chainManager.ts [hunk 1]
```diff
@@ -124,14 +124,35 @@ export default class ChainManager {
...
+        prompt: this.getEffectivePrompt(customModel),
+      });
       console.log(`Setting model to ${newModelKey}`);
     } catch (error) {
       console.error("createChainWithNewModel failed: ", error);
       console.log("modelKey:", newModelKey);
     }
   }
 
+  private getEffectivePrompt(customModel: CustomModel): ChatPromptTemplate {
```
Comment (logancyang) [inline src/LLMProviders/chainManager.ts:137]: Code is duplicated with other parts of this file. Please reorganize to be able to merge.

**Review by logancyang — CHANGES_REQUESTED** [review]
The change is ok but has duplication with existing parts of the file. Please reorganize and clean it up before merge.

**Comment by henryperkins** [comment]
Still running into connection error issues..

File: src/aiParams.ts [hunk 2]
```diff
@@ -73,6 +67,7 @@ export interface CustomModel {
   isBuiltIn?: boolean;
   enableCors?: boolean;
   core?: boolean;
+  azureOpenAIApiDeploymentName?: string; // Added for Azure OpenAI models
```
Comment (logancyang) [inline src/aiParams.ts]: Isn't this the same as the one above?

Comment (henryperkins) [reply src/aiParams.ts]: I'm just trying to remove the last 2 commits I made - struggling.

Comment (henryperkins) [reply src/aiParams.ts]: But to answer your question, currently when choosing custom model, you are not given a choice of selecting a model with your deployment ID. I might be wrong, but my testing has made me believe that several gpt-4o models are available on one azure openai deployment.

**Comment by rklasen** [comment]
Why was this closed? Is there another way to add `o1-preview` to Copilot?

**Pull request closed without merging**